### DISCUSSION TITLE
What has been the history of ED2 PFT numbers?

### DISCUSSION THREAD

**Discussion by dlebauer** [EDmodel/ED2#373] [Q&A] [answered]
Prior to ED2.2 https://github.com/EDmodel/ED2/commit/8fb76e517e77d1d0afc270de4fb17ceac75ee8c4#diff-2a3d35969c8f161509e3027c1fe96a6cc04c3e8abd8f17c0f1d679b41bd4a55d

The PFTs have been: 
|PFT id | [2012 rev-79](https://github.com/EDmodel/ED2/blame/rev-79/ED/run/ED2IN#L1156)  | [2017 no tag](https://github.com/EDmodel/ED2/commit/ff1d6aef4e4b749c21c1f781558e9245a7f7ec70/ED/run/ED2IN#L1356)   | [2019 v2.2.0 / current](https://github.com/EDmodel/ED2/blame/v.2.2.0/ED/run/ED2IN#L1601) |
|:---|:---|:---|:---|
|12       |  agricultural PFTs   | early savannah (thick bark)  | (Beta) Tropical broadleaf, early successional (thick bark) |
|13       |   agricultural PFTs  | mid savannah (thick bark)       | (Beta) Tropical broadleaf, mid-successional (thick bark) |
|14       | agricultural PFTs     | late savannah (thick bark)    | (Beta) Tropical broadleaf, late successional (thick bark) |
|15       | agricultural PFTs      |   | Araucaria  |

- What happened to the deprecated PFTs?
- Are there any records of the parameterizations for these PFTs (other than the source code, i.e., in tables or xml or h5 files)?

**Comment by mpaiao** [comment, marked as answer]
@dlebauer These agricultural PFTs were never implemented, they were just copies of the C4 grasses (PFT 1). We never had a truly agricultural PFT in ED2.

Araucaria used to be PFT 17, but was overwritten by the liana PFT, I just moved it to PFT 15 in one of the merges so it wouldn't be lost. The PFTs 12-15 between 2017 and 2019 PFTs are the same, I just gave them a lengthier, more informative name in 2019.
